api/gateway/v1alpha1/gateway_types.go

```diff
@@ -149,7 +163,10 @@ func (gw *Gateway) Validate(_ context.Context, _ kclient.Reader) error {
 			return fmt.Errorf("invalid neighbor IP %s: %w", neigh.IP, err)
 		}
 		if !neighIP.Is4() {
-			return fmt.Errorf("BGP neighbor %s IP %s must be an IPv4 address", neigh.IP, neigh.IP) //nolint:goerr113
+			return fmt.Errorf("BGP neighbor IP %s must be an IPv4 address", neigh.IP) //nolint:goerr113
+		}
+		if neighIP.IsMulticast() || neighIP.IsUnspecified() {
```

Comment:
Should we check that its not a loopback too?

Reply:
you can establish BGP session over loopback addresses, so it's not illegal. Whether it's something we will ever need/want to support in gateway, I do not know, but I did not want to break it just in case